app/src/main/java/org/openedx/app/AppActivity.kt

```diff
@@ -228,6 +228,7 @@ class AppActivity : AppCompatActivity(), InsetHolder, WindowSizeHolder {
 
     private fun handlePushNotification(data: Bundle) {
         val deepLink = DeepLink(data.toStringMap())
+        deepLink.notificationId?.let { viewModel.markNotificationAsRead(deepLink.notificationId) }
```

Comment:
We can move the null check inside the try-catch block of the view model. It will look neat.